boa/src/syntax/ast/node/template/mod.rs

```diff
@@ -0,0 +1,145 @@
+//! Template literal node.
+
+use super::Node;
+use crate::{builtins::Array, exec::Executable, value::Type, BoaProfiler, Context, Result, Value};
+use gc::{Finalize, Trace};
+
+#[cfg(feature = "deser")]
+use serde::{Deserialize, Serialize};
+use std::fmt;
+
+#[cfg(test)]
+mod tests;
+
+/// Template literals are string literals allowing embedded expressions.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///  - [MDN documentation][mdn]
+///
+/// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals
+/// [spec]: https://tc39.es/ecma262/#sec-template-literals
+#[cfg_attr(feature = "deser", derive(Serialize, Deserialize))]
+#[derive(Clone, Debug, Trace, Finalize, PartialEq)]
+pub struct TemplateLit {
+    elements: Vec<TemplateElement>,
+}
+
+impl TemplateLit {
+    pub fn new(elements: Vec<TemplateElement>) -> Self {
+        TemplateLit { elements }
+    }
+}
+
+impl Executable for TemplateLit {
+    fn run(&self, context: &mut Context) -> Result<Value> {
+        let _timer = BoaProfiler::global().start_event("TemplateLiteral", "exec");
+        let mut result = String::new();
+
+        for element in self.elements.iter() {
+            match element {
+                TemplateElement::String(s) => {
+                    result.push_str(s);
+                }
+                TemplateElement::Expr(node) => {
+                    let value = node.run(context)?;
+                    let s = value.to_string(context)?;
+                    result.push_str(&s);
+                }
+            }
+        }
+        Ok(result.into())
+    }
+}
+
+impl fmt::Display for TemplateLit {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "template literal")
```

Comment:
Here we should try to re-generate the original template, if we still have the information available.